scripts/perftune.py

```diff
@@ -738,21 +763,12 @@ def __max_rx_queue_count(self, iface):
         ixgbevf: VF NICs support up to 4 RSS queues.
         i40e:    PF NICs support up to 64 RSS queues.
         i40evf:  VF NICs support up to 16 RSS queues.
+        mlx:     PF/VF NICs throttle to 2 due to si problems when cpu is shared with scylla
 
         """
-        driver_to_max_rss = {'ixgbe': 16, 'ixgbevf': 4, 'i40e': 64, 'i40evf': 16}
-
-        driver_name = ''
-        ethtool_i_lines = run_read_only_command(['ethtool', '-i', iface]).splitlines()
-        driver_re = re.compile("driver:")
-        driver_lines = list(filter(lambda one_line: driver_re.search(one_line), ethtool_i_lines))
-
-        if driver_lines:
-            if len(driver_lines) > 1:
-                raise Exception("More than one 'driver:' entries in the 'ethtool -i {}' output. Unable to continue.".format(iface))
-
-            driver_name = driver_lines[0].split()[1].strip()
+        driver_to_max_rss = {'ixgbe': 16, 'ixgbevf': 4, 'i40e': 64, 'i40evf': 16, 'mlx4_en': 2,'mlx5_core' : 2}
```

Comment:
You seem to forget to actually call `ethtool -L` to reduce the number of RX queues.
Please, don't forget to test your code before sending a PR.

Reply:
yes, correct, this is was missed, ethtool -L
(and I tested IRQs but forgot to check number of queues)
will fix

Reply:
I used 8 and 32 as max for mellanox 4 and 5 drivers, this however should be validated with mellanox docs, I lack access, so I need help from @vladzcloudius or @eyalgutkind to help me validate these numbers (observed from VMs or from public docs, which are not definitive)